src/implementations/twig/components/accordion/accordion.story.js

```diff
@@ -43,8 +56,16 @@ export default {
   title: 'Components/Accordion',
 };
 
-export const Default = () => accordion(prepareAccordion(demoData));
-
+export const Default = (args) => accordion(prepareData(data, args));
+Default.argTypes = getArgTypes();
```

Comment:
Three, invoke `getArgTypes()` with a value for the input parameter. In the scope of the module `accordion.story.js` it's going to be `demoData`, therefore

```javascript
Default.argTypes = getArgTypes(demoData);
```

Inside the `getArgTypes`, `demoData` will be replaced with `data` and there will be no shadowing or scope creep.

The main benefit: you can extract the function out of the module at any time without any issues. But you also make the logic flow clear and do not risk accidental changes of the data.